### ISSUE TITLE
TODO: Get code coverage to 100%

### ISSUE THREAD

**Issue by DilumAluthge** [JuliaLinearAlgebra/Octavian.jl#11]
For reference, here is the current code coverage:

[![Code Coverage](https://codecov.io/gh/JuliaLinearAlgebra/Octavian.jl/branch/master/graph/badge.svg)](https://codecov.io/gh/JuliaLinearAlgebra/Octavian.jl)

**Comment by DilumAluthge** [comment]
This is done now.

**Comment by DilumAluthge** [comment]
Our coverage has fallen below 100%, so I'm reopening this.

**Comment by chriselrod** [comment]
There are a lot of branches based on the size of the arrays. Ideally, we'd be able to enable coverage for our tests over different array sizes.

**Comment by DilumAluthge** [comment]
I'm going to change the goal to 95% since I think that there are some lines that we'll never be able to cover.

**Comment by DilumAluthge** [comment]
@chriselrod We're at 40% now. Take a look at Codecov - do you think that we need to add more matrix sizes to `test/matmul_coverage.jl`? Or do we need to just start writing unit tests to cover the uncovered lines?

**Comment by chriselrod** [comment]
Two things that should be simple and help with this file:

1. Add a test dependency on something like StaticArrays.jl (or StrideArrays after it is registered). Testing with mutable statically sized arrays (meaning `StaticArrays.MArray`) would let us hit `inlineloopmul!`. It requires small statically sized arrays to be called. It should work out of the box with with StaticArrays, so that's worth testing to confirm itself.
2. Bigger matrices. In that file, you can see that none of the matrices were big enough to pack both `A` and `B`, so related code (i.e., `matmul_st_pack_A_and_B!`, `matmul_pack_A_and_B!`, and `sync_mul!` weren't tested.

**Comment by DilumAluthge** [comment]
> Bigger matrices. In that file, you can see that none of the matrices were big enough to pack both `A` and `B`, so related code (i.e., `matmul_st_pack_A_and_B!`, `matmul_pack_A_and_B!`, and `sync_mul!`) weren't tested.

https://github.com/JuliaLinearAlgebra/Octavian.jl/pull/49

**Comment by DilumAluthge** [comment]
> Add a test dependency on something like StaticArrays.jl (or StrideArrays after it is registered). Testing with mutable statically sized arrays (meaning `StaticArrays.MArray`) would let us hit `inlineloopmul!`. It requires small statically sized arrays to be called. It should work out of the box with with StaticArrays, so that's worth testing to confirm itself.

Sounds good. I'll make a PR to test with StaticArrays. A lot of people use StaticArrays for small matrices, so it's a good thing to test.

What size should our ``StaticArrays.MArray`s be? E.g. is 10-by-10 good?

**Comment by chriselrod** [comment]
> > Add a test dependency on something like StaticArrays.jl (or StrideArrays after it is registered). Testing with mutable statically sized arrays (meaning `StaticArrays.MArray`) would let us hit `inlineloopmul!`. It requires small statically sized arrays to be called. It should work out of the box with with StaticArrays, so that's worth testing to confirm itself.
> 
> Sounds good. I'll make a PR to test with StaticArrays. A lot of people use StaticArrays for small matrices, so it's a good thing to test.
> 
> What size should our ``StaticArrays.MArray`s be? E.g. is 10-by-10 good?

Yes, that's fine.

**Comment by DilumAluthge** [comment]
> Yes, that's fine. Note that Octavian should be a lot faster for 10x10:

Yeah, I remember that we start beating StaticArrays pretty early. What's the biggest matrix size at which StaticArrays is faster than Octavian?

**Comment by chriselrod** [comment]
> > Yes, that's fine. Note that Octavian should be a lot faster for 10x10:
> 
> Yeah, I remember that we start beating StaticArrays pretty early. What's the biggest matrix size at which StaticArrays is faster than Octavian?

![image](https://user-images.githubusercontent.com/8043603/105207318-601ebc80-5b15-11eb-80b0-ae73cc4906b7.png)

Oops, guess I should've started benchmarking at 1x1.
